### PR TITLE
code optimizations

### PR DESCRIPTION
**Pull request by adwk67**
- update library
- replace contains with covers
- keep lon, lat order consistent
- only add neighbour objects to KNN list when necessary

### CODE REVIEW

File: src/main/java/tech/stackable/gis/hbase/coprocessor/DistComp.java [hunk 1]
```diff
@@ -10,7 +10,7 @@ public class DistComp implements Comparator<Neighbor> {
     static final Log LOG = LogFactory.getLog(DistComp.class);
     private final Point2D origin;
 
-    public DistComp(double lat, double lon) {
```
Comment (razvan) [inline src/main/java/tech/stackable/gis/hbase/coprocessor/DistComp.java:13]: Now that `Neighbor` has the distance field, the `DistComp` doesn't need the origin or even compute the distance in `compare()` anymore.
Something like this should suffice:
```
    @Override
    public int compare(Neighbor o1, Neighbor o2) {
        if (o1.distance < 0 || o2.distance < 0)
            LOG.warn("negative distance detected!");
        return Double.compare(o1.distance, o2.distance);
    }
```

File: src/main/java/tech/stackable/gis/hbase/coprocessor/Neighbor.java [hunk 2]
```diff
@@ -4,10 +4,12 @@ public class Neighbor {
     public final String key;
     public final double lat;
     public final double lon;
+    public final double distance;
 
-    public Neighbor(String key, double lat, double lon) {
```
Comment (razvan) [inline src/main/java/tech/stackable/gis/hbase/coprocessor/Neighbor.java:8]: Now this class is structurally identical to the Point proto. Another (premature) optimization could be to get read of it and thus create less objects for the endpoint response.

**Review by razvan — COMMENTED** [review]
see comments

File: src/main/java/tech/stackable/gis/hbase/coprocessor/DistComp.java (hunk 1 above)
Comment (adwk67) [reply src/main/java/tech/stackable/gis/hbase/coprocessor/DistComp.java:13]: Done in c79e71a

File: src/main/java/tech/stackable/gis/hbase/coprocessor/Neighbor.java (hunk 2 above)
Comment (adwk67) [reply src/main/java/tech/stackable/gis/hbase/coprocessor/Neighbor.java:8]: good Point :) See [00e189f](https://github.com/stackabletech/hbase-gis/pull/7/commits/00e189f57bc36daec22172bdeae6ab7fe524909b)

**Review by razvan — APPROVED** [review]
lgtm

**Pull request merged**